src/core.c/CompUnit/PrecompilationRepository.pm6

```diff
@@ -233,14 +233,22 @@ class CompUnit::PrecompilationRepository::Default does CompUnit::PrecompilationR
         my $compiler-id = CompUnit::PrecompilationId.new-without-check($*RAKU.compiler.id);
         my $io = self.store.destination($compiler-id, $id);
         return False unless $io;
+
+        my $bc = "$io.bc".IO;
+        my $bc-lock = "$io.bc.lock".IO;
+
+        $bcl = $bc-lock.open(:w);
+        self.store.unlock;
```

Comment:
Shouldn't the lock on the whole store be obsoleted by the lock on the individual precomp file?

We shouldn't add more assumptions that precomp files are stored in a directory structure as files. Any change should get us away from that instead. So we should either have the precomp store manage the locking of the individual precomp file, or put the lock in a temp directory. Since the lock will only be useful if it's name is predictable (so other rakudo processes will use the same file), having the precomp store manage locking seems preferable. That would also fit well with the existing API (the lock getting taken implicitly by `self.store.destination` which already gets all the information it needs to lock the individual precomp file.

Reply:
This is true, but that lock file might not yet exist. So locking the store, to create the file just made sense. I do agree that this is something that the store might be better at, however.